add delta view to taplo-cli

drahnr opened the pull request:
It's not uncommon to use a formatter in CI where a subset of contributors do not have the tool installed. That set of contributors is usually very appreciative of direct cues what to fix.

The changes:

Adds a dependency to `prettydiff` and `ansi_term` for creating the diff and coloring, there is currently (almost) no clustering of diff lines, this is to be improved.

tamasfe commented on crates/taplo-cli/src/commands/format.rs:
```diff
@@ -65,6 +66,92 @@ impl<E: Environment> Taplo<E> {
 
+    fn print_diff(path: impl AsRef<Path>, original: &str, formatted: &str) {
+        let path = path.as_ref();
+        println!("diff a/{path} b/{path}", path = path.display());
```
The stdio is behind the `Environment` abstraction in order to support WASM. The standard output should be available here via `self.env.stdout()`, all printing should be done via that instead.

drahnr replied:
Thanks, I wasn't aware

tamasfe commented:
I think this could be gated behind a flag like `--diff` as this can get very verbose, looks good otherwise, thanks!

drahnr commented:
Added `--diff` flag, both for `--check` and non-check. Introduced a scope macro `echo!` that utilizes `self.env`. I think it's ready for another review pass :)

tamasfe commented:
Looks great, thank you!
